equipment_handler/source/tmp421.c

```diff
@@ -338,23 +338,26 @@ int tmp421_detect(uint8_t sadd) {
         kind = tmp421;
         break;
     case TMP422_DEVICE_ID:
-        //		if (addr == 0x2a)
+#if 0
+        if (addr == 0x2a)
+            kind = tmp422;
+#endif
         return 1;
-        kind = tmp422;
-        break;
     case TMP423_DEVICE_ID:
-        //		if (addr != 0x4c && addr != 0x4d)
+#if 0
+        if (addr != 0x4c && addr != 0x4d)
+            kind = tmp423;
+#endif
         return 1;
-        kind = tmp423;
-        break;
     case TMP441_DEVICE_ID:
         kind = tmp441;
         break;
     case TMP442_DEVICE_ID:
-        //		if (addr != 0x4c && addr != 0x4d)
+#if 0
```

Comment:
What's this for?

Reply:
The compiler was complaining about unreachable statements, because we had "return 1" followed by "break". It looked to me like the commented out stuff was valid tests for the devices in question but we didn't expect those devices, so we decided to fail (return) instead of succeed. I used the #if 0 to illustrate the control we could use and have the tests succeed